pkg/memtier/proc.go

```diff
@@ -500,34 +507,81 @@ func (f *ProcPagemapFile) read(readData *[]byte, readBuf *[]byte, pagemapOffset
 	return nbytes
 }
 
-func pagemapBitsSatisfied(pagemapBits uint64,
-	pageMustBePresent, pageMustNotBePresent,
-	pageMustBeExclusive, pageMustNotBeExclusive,
-	pageMustBeDirty, pageMustNotBeDirty bool) bool {
+func isPagePresent(pagemapBits uint64, pageMustBePresent, pageMustNotBePresent bool) bool {
```

Comment:
This function returns `true` when expectation on the "present" pagemap bit is violated: either the bit is expected to be false but it is true, or the other way around. The function should be named accordingly. "isPagePresent" sounds like the function returns true if the page has the present bit set. How about "violatesPresentBit"?

A real simplification could be done to the code by adding only one function instead of three:
`func pagemapBitViolated(pagemapBits uint64, bit uint64, mustBeTrue, mustBeFalse bool) bool`
and then call this function three times:
```
if pagemapBitViolated(pagemapBits, PM_PRESENT, pageMustBePresent, pageMustNotBePresent) {
    return false
}
```